File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/builders/widgets/AbstractWidgetBuilder.java
package cz.cvut.fel.matyapav.afandroid.builders.widgets;

import android.app.Activity;
import android.content.Context;
import android.view.View;

import cz.cvut.fel.matyapav.afandroid.components.parts.AFField;

/**
 * Created by Pavel on 14.02.2016.
 */
public interface AbstractWidgetBuilder {

    View buildFieldView(Context context);

    void setData(AFField field, Object value);

    Object getData(AFField field);
}

File: AFMobile/AFWinPhone/AFWinPhone/builders/widgets/AbstractWidgetBuilder.cs
using AFWinPhone.components.parts;
using System;
using Windows.UI.Xaml;

namespace AFWinPhone.builders.widgets
{
    interface AbstractWidgetBuilder
    {

        FrameworkElement buildFieldView();

        void setData(AFField field, Object value);

        Object getData(AFField field);
    }
}
